aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/input/InputStepTest.java

```diff
@@ -65,4 +74,77 @@ public void parameter() throws Exception {
         System.out.println(b.getLog());
         assertTrue(b.getLog().contains("after: false"));
     }
+
+    @Test
+    @Issue("JENKINS-26363")
+    public void test_cancel_run_by_input() throws Exception {
+        JenkinsRule.WebClient webClient = j.createWebClient();
+        JenkinsRule.DummySecurityRealm dummySecurityRealm = j.createDummySecurityRealm();
+        GlobalMatrixAuthorizationStrategy authorizationStrategy = new GlobalMatrixAuthorizationStrategy();
+
+        j.jenkins.setSecurityRealm(dummySecurityRealm);
+
+        // Only give "alice" basic privs. That's normally not enough to Job.CANCEL, only for the fact that "alice"
+        // is listed as the submitter.
+        addUserWithPrivs("alice", authorizationStrategy);
+        // Only give "bob" basic privs.  That's normally not enough to Job.CANCEL and "bob" is not the submitter,
+        // so they should be rejected.
+        addUserWithPrivs("bob", authorizationStrategy);
+        // Give "charlie" basic privs + Job.CANCEL.  That should allow user3 cancel.
+        addUserWithPrivs("charlie", authorizationStrategy);
+        authorizationStrategy.add(Job.CANCEL, "charlie");
+
+        j.jenkins.setAuthorizationStrategy(authorizationStrategy);
+
+        final WorkflowJob foo = j.jenkins.createProject(WorkflowJob.class, "foo");
+        ACL.impersonate(User.get("alice").impersonate(), new Runnable() {
+            @Override
+            public void run() {
+                foo.setDefinition(new CpsFlowDefinition("input id: 'InputX', message: 'OK?', ok: 'Yes', submitter: 'alice'"));
```

Comment:
BTW this `impersonate` and granting `RUN_SCRIPTS` would probably be unnecessary if you just passed `true` for `sandbox`.